src/main.rs

```diff
@@ -1,3 +1,192 @@
+            .collect::<Vec<String>>()
+            .join(",");
+        let fields = self
+            ._fields
+            .into_iter()
+            .map(|(field, value)| format!("{field}={value}", field = field, value = value))
+            .collect::<Vec<String>>()
+            .join(",");
+
+        format!(
+            "measurement,{tags} {fields} time",
+            tags = tags,
+            fields = fields
+        )
+    }
+}
+
+pub struct InfluxDbClient {
+    _url: String,
+    _database: String,
+    // _auth: InfluxDbAuthentication | NoAuthentication
+}
+
+pub fn main() {}
+
+impl InfluxDbClient {
+    pub fn ping(&self) -> impl Future<Item = (String, String), Error = ()> {
+        Client::new()
+            .get(format!("{}/ping", self._url).as_str())
+            .send()
+            .map(|res| {
+                let build = res
+                    .headers()
+                    .get("X-Influxdb-Build")
+                    .unwrap()
+                    .to_str()
+                    .unwrap();
+                let version = res
+                    .headers()
+                    .get("X-Influxdb-Version")
+                    .unwrap()
+                    .to_str()
+                    .unwrap();
+
+                (String::from(build), String::from(version))
+            })
+            .map_err(|err| println!("request error: {}", err))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{InfluxDbClient, InfluxDbQuery};
+    use tokio::runtime::current_thread::Runtime;
+
+    fn get_runtime() -> Runtime {
+        Runtime::new().expect("Unable to create a runtime")
+    }
+
+    fn create_client() -> InfluxDbClient {
+        InfluxDbClient {
+            _url: String::from("http://localhost:8086"),
+            _database: String::from("test"),
+        }
+    }
+
+    #[test]
+    fn test_ping() {
+        let client = create_client();
+        let result = get_runtime().block_on(client.ping());
+        assert!(!result.is_err(), "Should be no error");
+
+        let (build, version) = result.unwrap();
+        assert!(!build.is_empty(), "Build should not be empty");
+        assert!(!version.is_empty(), "Build should not be empty");
+
+        println!("build: {} version: {}", build, version);
+    }
+
+    #[test]
+    fn test_write_builder_single_field() {
+        let mut query = InfluxDbQuery::write();
+
+        query.add_field("water_level", "2");
+        assert_eq!(query.build(), "measurement, water_level=2 time");
+    }
+
+    #[test]
+    fn test_write_builder_multiple_fields() {
+        let mut query = InfluxDbQuery::write();
+
+        query.add_field("water_level", "2");
+        query.add_field("boat_count", "31");
+        query.add_field("algae_content", "0.85");
+        assert_eq!(
+            query.build(),
+            "measurement, water_level=2,boat_count=31,algae_content=0.85 time"
+        );
+    }
+
+    // fixme: double space
+    // fixme: quoting / escaping of long strings
```

Comment:
Don't add _fixme_ in tests. If the code doesn't do what you want, the test should fail. That's the whole idea about testing.